consensus/hotstuff/consumer.go

```diff
@@ -234,3 +237,31 @@ type TimeoutCollectorConsumer interface {
 	// and must handle repetition of the same events (with some processing overhead).
 	OnNewTcDiscovered(certificate *flow.TimeoutCertificate)
 }
+
+// CommunicatorConsumer consumes outbound notifications produced by HotStuff and it's components.
+// Notifications allow the HotStuff core algorithm to communicate with the other actors of the consensus process.
+// Implementations must:
+//   - be concurrency safe
+//   - be non-blocking
+//   - handle repetition of the same events (with some processing overhead).
+type CommunicatorConsumer interface {
+	// SendVote notifies about intent to send a vote for the given parameters to the specified recipient.
+	// Prerequisites:
+	// Implementation must be concurrency safe; Non-blocking;
+	// and must handle repetition of the same events (with some processing overhead).
+	SendVote(blockID flow.Identifier, view uint64, sigData []byte, recipientID flow.Identifier)
```

Comment:
I like the suggestion to stay consistent with our existing naming convention. Jordan's naming proposal looks good to me. Additional thoughts:
* I am not sure, but it might improve clarity if we reflect in the naming that these are our _own_ messages `OnOwnVote`, `OnOwnProposal`, `OnOwnTimeout` (because a consensus participants might also relay blocks from other nodes for example). I feel these are most generic, without implying what should happen with the event. 
I feel it is more the responsibility of the consumer to decide what to do with the events. So for the messages to reach other nodes, there needs to be a specific consumer (the `MessageHub`) that relays those.
* We already have an event `OnVoting` in the `hotstuff.Consumer` which I feel is now duplicated.